cumulusci/robotframework/Salesforce.py

```diff
@@ -24,9 +25,25 @@
 lex_locators = {}  # will be initialized when Salesforce is instantiated
 
 
-class RandomStringGenerator:
+class StringGenerator:
+    def __init__(self, func):
+        self.func = func
+
     def __str__(self):
-        return String().generate_random_string()
+        return self.func()
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
+
+
+class FakerTemplateLibrary:
+    faker = faker.Faker()
+
+    def __getattr__(self, name):
+        # defer loading heavy faker library until actually needed
```

Comment:
Looks like despite this comment it's initialized at import time on line 40. Or do you mean that faker defers loading subsets of itself until they are used?

Reply:
I think I refactored the laziness away at some point but I've added it back now because Faker is big and I'm not sure how/when it initializes all of its internal databases. It's just a couple of lines of code to make it a lazy load.

Reply:
Done.

Reply:
Are you certain this code buys us anything? I would be surprised to see the overhead being more than a few ms.

Reply:
No, I'm not certain, but I do know that 

a) Faker is about 4 MB of Python code including everything from German names to Dutch company names. It may do lazy loading internally but I'm not sure and I was hoping not to do a lot of investigation just to save the one line of code.

b) its difficult to benchmark because once a module is loaded, its loaded, so running tests is difficult. And even from a fresh Python its very subject to caching effects.

c) CCI seems to have a lot of "technical debt" relating to performance which I am reluctant to add to. It takes 3 seconds just to get a command listing from CCI on my (cold) laptop and 1 second if its hot.

d) its just one line of code. I don't think it is time well spent to do a bunch of investigation when I can be confident that I'm not adding to CCI's performance technical debt with the one line of code.